cleanlab/internal/multiannotator_utils.py

```diff
@@ -134,3 +150,122 @@ def format_multiannotator_labels(labels: LabelLike) -> Tuple[pd.DataFrame, dict]
     formatted_labels = labels.replace(label_map)
 
     return formatted_labels, inverse_map
+
+
+def check_consensus_label_classes(
+    labels_multiannotator: pd.DataFrame,
+    consensus_label: np.ndarray,
+    consensus_method: str,
+) -> None:
+    """Check if any classes no longer appear in the set of consensus labels (established using the consensus_method stated)"""
+    unique_ma_labels = np.unique(labels_multiannotator.replace({pd.NA: np.NaN}).astype(float))
+    unique_ma_labels = unique_ma_labels[~np.isnan(unique_ma_labels)]
+    labels_set_difference = set(unique_ma_labels) - set(consensus_label)
+
+    if len(labels_set_difference) > 0:
+        print(
+            "CAUTION: Number of unique classes has been reduced from the original data when establishing consensus labels"
+            f"using consensus method '{consensus_method}', likely due to some classes being rarely annotated."
+            "If training a classifier on these consensus labels, it will never see any of the omitted classes unless you"
+            "manually replace some of the consensus labels."
+            f"Classes in the original data but not in consensus labels: {list(map(int, labels_set_difference))}"
+        )
+
+
+def compute_soft_cross_entropy(
+    labels_multiannotator: pd.DataFrame,
+    pred_probs: np.ndarray,
+) -> float:
+    """Compute soft cross entropy between the annotators' empirical label distribution and model pred_probs"""
+    num_classes = get_num_classes(pred_probs=pred_probs)
+
+    empirical_label_distribution = np.full((len(labels_multiannotator), num_classes), np.NaN)
+    for i in range(len(labels_multiannotator)):
+        s = labels_multiannotator.iloc[i]
+        empirical_label_distribution[i, :] = value_counts(
+            s.dropna(), num_classes=num_classes
+        ) / len(s.dropna())
+
+    clipped_pred_probs = np.clip(pred_probs, a_min=1e-6, a_max=None)
```

Comment:
note here you still have 1e-6 instead of the smaller 1e-30.

I think it would be safest to define a constant variable = 1e-30 at the top of this file and then refer to this constant variable everywhere